pkg/controller/appmetricsservice/appmetricsservice_controller.go

```diff
@@ -332,6 +396,60 @@ func (r *ReconcileAppMetricsService) Reconcile(request reconcile.Request) (recon
 	}
 	//#endregion
 
+	//#region Backups
+	if len(instance.Spec.Backups) > 0 {
+		backupjobSA := &corev1.ServiceAccount{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: "backupjob", Namespace: instance.Namespace}, backupjobSA)
+		if err != nil {
+			reqLogger.Error(err, "A 'backupjob' ServiceAccount is required for the requested backup CronJob(s). Will check again in 10 seconds")
+			return reconcile.Result{RequeueAfter: time.Second * 10}, nil
+		}
+	}
+
+	existingCronJobs := &batchv1beta1.CronJobList{}
+	opts = client.InNamespace(instance.Namespace).MatchingLabels(labels(instance, "backup"))
+	err = r.client.List(context.TODO(), opts, existingCronJobs)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	desiredCronJobs, err := backups(instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	for _, desiredCronJob := range desiredCronJobs {
+		if err := controllerutil.SetControllerReference(instance, &desiredCronJob, r.scheme); err != nil {
```

Comment:
Why not add the owner when the object is created as it is done in the Getting started example? 
SHows that you arre creating the objects then after looking for them to update with this info. Could it not cause issues scenarios as for example update an object that was not actually created/managed for here?

Reply:
The `backups` function is just a helper for the moment to create the structure of the CronJob objects. It doesn't have all of the context to be able to set the owner reference for now, and we don't want to make it more complex for the moment. I've got some ideas for ways to re-structure the entire controller logic to make it more comprehensible, and this will change as part of that. We don't currently have the cycles to try that out though, so that will be for a future release.

It won't cause any problems here, since `desiredCronJobs` here is just the CronJob structs created based on the CR `instance`, and we're just setting the owner reference before doing an `r.client.Create` or `r.client.Update`.

Reply:
But the owner is part of the obj. Sincerely, IMHO that it is very more complex and hard to understand/mantain that just do as it is done in the Getting started example or [here](https://github.com/aerogear/mobile-security-service-operator/blob/master/pkg/controller/mobilesecurityservicebackup/cronjobs.go#L72) for example. However,  it is up to you folks,